### PR TITLE
[multireg] Do not compact if multireg field is not homogenous

### PR DESCRIPTION
**Pull request by tjaychen**
Related to #1872

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
Not really sure this is the right fix...at least when i tried it with an error case lint no longer broke.

File: util/reggen/validate.py [hunk 1]
```diff
@@ -994,13 +994,21 @@ def validate_multi(mreg, offset, addrsep, width, top):
...
         while bpos < width:
-            trypos = bused << bpos
-            if trypos > max_rval:
+            # bused is a bit mask of how many bits the fields need
+            # bpos is the current LSB of the bits allocated for the fields
+            trypos = bused << bpos;
+
+            # if register can no longer contain another homogenous field
+            # if the field is not homongenous and has been allocated
+            # Currently we only compact if field is homongenous
```
Comment (msfschaffner) [inline util/reggen/validate.py:1003]: nit: two instances of homo**n**genous

Comment (tjaychen) [reply util/reggen/validate.py:1003]: o boy, thanks for catching that.

**Comment by tjaychen** [comment]
thanks for reviewing @msfschaffner 
@eunchan let me know what you think.  This fix is clearly a hack...i think the multireg portion (at least in validate.py) probably deserves a re-write at some point.. so this is a band-aid at best..

**Comment by eunchan** [comment]
> thanks for reviewing @msfschaffner
> @eunchan let me know what you think. This fix is clearly a hack...i think the multireg portion (at least in validate.py) probably deserves a re-write at some point.. so this is a band-aid at best..

It is a hack but I like this too. I actually REALLY like this. Simple but solves everything :)

**Comment by eunchan** [comment]
BTW, will you re-create the IP registers ? Or nothing need to be changed with this?

**Comment by tjaychen** [comment]
> BTW, will you re-create the IP registers ? Or nothing need to be changed with this?

thanks for the reminder. I just ran a make regs and looks like nothing changed.

**Pull request merged**